src/NextGen/DonationForm/Repositories/DonationFormRepository.php

```diff
@@ -28,161 +26,262 @@ class DonationFormRepository {
+    public function prepareQuery(): ModelQueryBuilder
     {
-        if ($block->name === "custom-block-editor/donation-amount-levels") {
-            $node = Amount::make('amount')
-                ->levels(...array_map('absint', $block->attributes->levels))
-                ->allowCustomAmount()
-                ->defaultValue(50)
-                ->required();
-        } elseif ($block->name === "custom-block-editor/donor-name") {
-            $node = Name::make('name')->tap(function ($group) use ($block) {
-                $group->getNodeByName('firstName')
-                    ->label($block->attributes->firstNameLabel)
-                    ->placeholder($block->attributes->firstNamePlaceholder);
-
-                $group->getNodeByName('lastName')
-                    ->label($block->attributes->lastNameLabel)
-                    ->placeholder($block->attributes->lastNamePlaceholder)
-                    ->required($block->attributes->requireLastName);
-
-                if ($block->attributes->showHonorific) {
-                    $group->getNodeByName('honorific')
-                        ->label('Title')
-                        ->options(...$block->attributes->honorifics);
-                } else {
-                    $group->remove('honorific');
-                }
-            });
-        } elseif ($block->name === "custom-block-editor/paragraph") {
-            $node = Paragraph::make(substr(md5(mt_rand()), 0, 7))
-                ->content($block->attributes->content);
-        } elseif ($block->name === "custom-block-editor/email-field") {
-            $node = Email::make('email')->emailTag('email');
-        } elseif ($block->name === "custom-block-editor/payment-gateways") {
-            $node = PaymentGateways::make('gatewayId');
-        } elseif ($block->name === "custom-block-editor/donation-summary") {
-            $node = DonationSummary::make('donation-summary');
-        } elseif ($block->name === "custom-block-editor/company-field") {
-            $node = Text::make('company');
-        } else {
-            $node = Text::make($block->clientId);
-        }
+        $builder = new ModelQueryBuilder(DonationForm::class);
 
-        if ('field' === $node->getNodeType()) {
-            // Label
-            if (property_exists($block->attributes, 'label')) {
-                $node->label($block->attributes->label);
-            }
+        return $builder->from('posts')
+            ->select(
+                ['ID', 'id'],
+                ['post_date', 'createdAt'],
+                ['post_modified', 'updatedAt'],
+                ['post_status', 'status'],
+                ['post_title', 'formTitle'],
+                ['post_content', 'blocksContent']
+            )
+            ->attachMeta(
+                'give_formmeta',
+                'ID',
+                'form_id',
+                ...DonationFormMetaKeys::getColumnsForAttachMetaQuery()
+            )
+            ->where('post_type', 'give_forms');
```

Comment:
I think we should add something like where `post_content` is not null to differentiate legacy & next gen forms.

Reply:
I worry that this is too ambiguous a way of differentiating. Is there something more explicit we can check?

Reply:
The specific differences in the database are the `post_content` column and the `formBuiderSettings` meta (which needs to be renamed).

Reply:
sounds good